frontend/simple/utils/giLodash.js

```diff
@@ -12,3 +12,169 @@ export function mapValues (obj: Object, fn: Function, o: Object = {}) {
 export function cloneDeep (obj: JSONObject) {
   return JSON.parse(JSON.stringify(obj))
 }
+
+function isMergeableObject (val) {
+  var nonNullObject = val && typeof val === 'object'
+  return nonNullObject && Object.prototype.toString.call(val) !== '[object RegExp]' && Object.prototype.toString.call(val) !== '[object Date]'
+}
+
+export function merge (obj: Object, src: Object) {
+  for (let key in src) {
+    if (isMergeableObject(obj[key])) {
+      if (isMergeableObject(src[key])) {
+        let clone = cloneDeep(src[key])
+        merge(obj[key], clone)
+      } else {
+        obj[key] = src[key]
+      }
+    } else {
+      if (isMergeableObject(src[key])) {
+        let clone = cloneDeep(src[key])
+        obj[key] = clone
+      } else {
+        obj[key] = src[key]
+      }
+    }
+  }
+}
```

Comment:
wouldn't it be nicer to flatten out these conditions?

```
for (let key in src) {
    if (isMergeableObject(obj[key])) {
      if (isMergeableObject(src[key])) {
        let clone = cloneDeep(src[key])
        merge(obj[key], clone)
        continue
      } 
      obj[key] = src[key]
      continue
    }
    if (isMergeableObject(src[key])) {
      let clone = cloneDeep(src[key])
      obj[key] = clone
      continue
    }
    obj[key] = src[key]
  }
```

Reply:
This I can do